src/components/NewMessageModal.vue

```diff
@@ -36,158 +33,117 @@ export default {
+					...data,
+					body: data.isHtml ? data.body.value : toPlain(data.body).value,
+				}
+				await this.$store.dispatch('outbox/updateMessage', { message: dataForServer, id: this.composerData.id })
+			} else {
+				if (data.draftId === undefined && this.draft) {
+					logger.debug('draft data does not have a draftId, adding one', {
+						draft: this.draft,
+						data,
+						id: this.draft.databaseId,
+					})
+					data.draftId = this.draft.databaseId
+				}
+				const dataForServer = {
+					...data,
+					to: data.to.map(this.recipientToRfc822).join(', '),
+					cc: data.cc.map(this.recipientToRfc822).join(', '),
+					bcc: data.bcc.map(this.recipientToRfc822).join(', '),
+					body: data.isHtml ? data.body.value : toPlain(data.body).value,
+				}
+				const { id } = await saveDraft(data.account, dataForServer)
 
-			// Remove old draft envelope
-			this.$store.commit('removeEnvelope', { id: data.draftId })
-			this.$store.commit('removeMessage', { id: data.draftId })
+				// Remove old draft envelope
+				this.$store.commit('removeEnvelope', { id: data.draftId })
+				this.$store.commit('removeMessage', { id: data.draftId })
 
-			// Fetch new draft envelope
-			await this.$store.dispatch('fetchEnvelope', id)
+				// Fetch new draft envelope
+				await this.$store.dispatch('fetchEnvelope', id)
 
-			return id
+				return id
+			}
 		},
 		async sendMessage(data) {
 			logger.debug('sending message', { data })
-			const now = new Date().getTime()
-			const dataForServer = {
-				accountId: data.account,
-				sendAt: Math.floor(now / 1000), // JS timestamp is in milliseconds
-				subject: data.subject,
-				body: data.isHtml ? data.body.value : toPlain(data.body).value,
-				isHtml: data.isHtml,
-				isMdn: false,
-				inReplyToMessageId: '',
-				to: data.to,
-				cc: data.cc,
-				bcc: data.bcc,
-				attachmentIds: [],
-			}
-			const message = await this.$store.dispatch('outbox/enqueueMessage', {
-				message: dataForServer,
-			})
-
-			await this.$store.dispatch('outbox/sendMessage', { id: message.id })
-
-			// Remove old draft envelope
-			this.$store.commit('removeEnvelope', { id: data.draftId })
-			this.$store.commit('removeMessage', { id: data.draftId })
-		},
-		async fetchOriginalMessage() {
-			if (this.templateMessageId === undefined) {
-				this.fetchingTemplateMessage = false
-				return
-			}
-			this.loading = true
-			this.error = undefined
-			this.errorMessage = ''
-
-			logger.debug(`fetching original message ${this.templateMessageId}`)
+			if (this.composerMessage.type === 'outbox') {
+				const now = new Date().getTime()
+				const dataForServer = {
+					accountId: data.account,
+					sendAt: Math.floor(now / 1000), // JS timestamp is in milliseconds
+					subject: data.subject,
+					body: data.isHtml ? data.body.value : toPlain(data.body).value,
+					isHtml: data.isHtml,
+					isMdn: false,
+					inReplyToMessageId: '',
+					to: data.to,
+					cc: data.cc,
+					bcc: data.bcc,
+					attachmentIds: [],
```

Comment:
this should be attachments I think

Reply:
yes but no. The old ``\OCA\Mail\Controller\AccountsController::send`` took a nested array `attachments`. ``\OCA\Mail\Controller\OutboxController::create`` takes an int array `attachmentIds`. I don't know how to bridge those.